src/OrleansGoogleUtils/Providers/Streams/GooglePubSubAdapterFactory.cs

```diff
@@ -0,0 +1,97 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using Orleans.Providers;
+using Orleans.Providers.Streams.Common;
+using Orleans.Runtime;
+using Orleans.Streams;
+using System;
+using System.Threading.Tasks;
+
+namespace Orleans.Serialization.Providers.Streams
+{
+    public class GooglePubSubAdapterFactory<TDataAdapter> : IQueueAdapterFactory
+        where TDataAdapter : IGooglePubSubDataAdapter
+    {
+        private string _projectId;
+        private string _topicId;
+        private string _deploymentId;
+        private string _providerName;
+        private int _cacheSize;
+        private int _numSubscriptions;
+        private TimeSpan? _deadline;
+        private HashRingBasedStreamQueueMapper _streamQueueMapper;
```

Comment:
This will limit GooglePubSubStreamProvider only be able to use HashRingBasedStreamQueueMapper as its queue mapper. Have you considered add a QueueMapperFactory here to make `IStreamQueueMapper `configurable?